src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -94,7 +95,62 @@ object Content {
   def fromLeafFiles(files: Seq[FileStatus]): Content = Content(Directory.fromLeafFiles(files))
 }
 
-case class Directory(name: String, files: Seq[FileInfo] = Seq(), subDirs: Seq[Directory] = Seq())
+/**
+ * Directory is a representation of file system directory. It consists of a name (directory name),
+ * a list of files represented by sequence of [[FileInfo]], and a list of subdirectories.
+ *
+ * @param name Directory name.
+ * @param files List of leaf files in this directory.
+ * @param subDirs List of sub-directories in this directory.
+ */
+case class Directory(
+    name: String,
+    files: Seq[FileInfo] = Seq(),
+    subDirs: Seq[Directory] = Seq()) {
+  /**
+   * Merge two Directory objects. For e.g., merging the following directories
+   * /file:/C:/
+   *          a/
+   *            b/
+   *              f1, f2
+   * and
+   * /file:/C:/
+   *          a/
+   *            f3, f4
+   * will be
+   * /file:/C:/
+   *           a/
+   *             f3, f4
+   *             b/
+   *               f1, f2
+   *
+   * @param that The other directory to merge this with.
+   * @return Merged directory
+   * @throws HyperspaceException To merge two directories, their name should be same.
```

Comment:
`@throws HyperspaceException if two directories to merge have different names.`

Reply:
thanks @imback82 , added